docs/build/guides/transactions/clawbacks.mdx

```diff
@@ -44,354 +44,865 @@ Here we’ll cover the following approaches to clawing back an asset.
+  }
+}
+
+// Fund accounts first
+function fundAccounts() {
+  console.log("=== FUNDING ACCOUNTS WITH XLM ===");
+  return Promise.all([
+    server.requestAirdrop(A.publicKey()),
+    server.requestAirdrop(B.publicKey()),
+    server.requestAirdrop(C.publicKey()),
+  ]).then(() => {
+    console.log("All accounts funded with XLM via airdrop");
+    // Wait for funding to complete
+    return new Promise((resolve) => setTimeout(resolve, 3000));
+  });
+}
+
+// Enables AuthClawbackEnabledFlag on an account.
 function enableClawback(account, keys) {
-  return server.submitTransaction(
+  console.log(
+    `Enabling clawback flags on account ${formatAccount(account.accountId())}`,
+  );
+  return submitAndPollTransaction(
     buildTx(account, keys, [
       sdk.Operation.setOptions({
         setFlags: sdk.AuthClawbackEnabledFlag | sdk.AuthRevocableFlag,
       }),
     ]),
+    "Enable Clawback Flags",
   );
 }
 
-/// Establishes a trustline for `recipient` for ASSET (from above).
+// Establishes a trustline for `recipient` for the CLAW Asset
 const establishTrustline = function (recipient, key) {
-  return server.submitTransaction(
+  console.log(
+    `${formatAccount(recipient.accountId())} establishing trustline for ${
+      ASSET.code
+    }`,
+  );
+  return submitAndPollTransaction(
     buildTx(recipient, key, [
       sdk.Operation.changeTrust({
         asset: ASSET,
         limit: "5000", // arbitrary
       }),
     ]),
+    `Establish Trustline (${formatAccount(recipient.accountId())})`,
   );
 };
 
-/// Retrieves latest account info for all accounts.
+// Retrieves latest account info for all accounts.
 function getAccounts() {
   return Promise.all([
-    server.loadAccount(A.publicKey()),
-    server.loadAccount(B.publicKey()),
-    server.loadAccount(C.publicKey()),
+    server.getAccount(A.publicKey()),
+    server.getAccount(B.publicKey()),
+    server.getAccount(C.publicKey()),
   ]);
 }
 
-/// Enables clawback on A, and establishes trustlines from C, B -> A.
-function preamble() {
-  return getAccounts().then(function (accounts) {
-    let [accountA, accountB, accountC] = accounts;
-    return enableClawback(accountA, A).then(
-      Promise.all([
-        establishTrustline(accountB, B),
-        establishTrustline(accountC, C),
-      ]),
-    );
-  });
+// Show XLM balances (after funding)
+function showXLMBalances(accounts) {
+  console.log("\n=== XLM BALANCES ===");
+  return Promise.all(
+    accounts.map((acc) => {
+      return getXLMBalance(acc.accountId()).then((balance) => {
+        console.log(`${formatAccount(acc.accountId())}: ${balance} XLM`);
+      });
+    }),
+  );
 }
 
-/// Helps simplify creating & signing a transaction.
+// Get XLM balance using account ledger entry
+function getXLMBalance(accountId) {
+  return server
+    .getAccountEntry(accountId)
+    .then((accountEntry) => {
+      return (parseInt(accountEntry.balance().toString()) / 10000000).toFixed(
```

Comment:
I hate counting zeroes :smile: 
```suggestion
      return (parseInt(accountEntry.balance().toString()) / 1e7).toFixed(
```

Reply:
Also we should be careful with the `xdr.Int64` math here. Annoyingly, you [need to upscale](https://stackoverflow.com/a/54409977) first to avoid the `parseInt` call truncating large values. I recommend a helper like:

```typescript
function scaleAsset(x: bigint): float {
    return Number((x * 10n) / 10000000n) / 10; // one decimal place
}
```

and passing `accountEntry.balance().toBigInt()` to this, instead.